include/curl/easy.h

```diff
@@ -25,6 +25,67 @@
 extern "C" {
 #endif
 
+/* Application can use a memory PKCS12 certificate with CURLOPT_SSLCERT
+ *  with OpenSSL, Schannel or SecTransport
+ *
+ *
+ *  Example if with certificate binary data and size in:
+ *    void* certdata;
+ *    size_t certsize;
+ *    ...
+ *  set the certificate with:
+ *    struct curl_blob structblob;
+ *    curl_init_blob_dup(&structblob, certdata, certsize);
+ *    curl_easy_setopt(curl, CURLOPT_KEYPASSWD, &structblob);
+ *             my_setopt_str(curl, CURLOPT_SSLCERT, &structblob);
+ *  struct curl_blob is just a structures which contain pointer to
+ *  data buffer, length and a boolean flag about buffer duplication
+ *
+ *  with curl_init_blob_dup, certdata can be discarded after
+ *    calling curl_easy_setopt
+ *  with curl_init_blob_persist, certdata need be valid when the
+ *  connexion is made
+ *
+ * Note : never call curl_init_blob_xxx without a pointer to a
+ *   valid curl_blob
+ *
+ *
+ * about the idea, see https://curl.haxx.se/mail/lib-2016-09/0074.html */
+
+
+#define CURL_BLOB_DUPFLAG_COPY     (1)
+#define CURL_BLOB_DUPFLAG_NOCOPY   (0)
+#define CURL_BLOB_DUPFLAG_COPYMASK (1)
+
+/* the struct curl_blob store binary data parameters
+ * the structure size is 0x20 and can be followed by binary data */
+
+struct curl_blob {
+    void *data;
+    size_t len;
+    size_t flags;
+};
+
+/*
+ * Macro WHILE_FALSE may be used to build single-iteration do-while loops,
+ * avoiding compiler warnings. Mostly intended for other macro definitions.
+ */
+
+#define curl_init_blob_flag(structblob, ptr, init_len, init_flags) \
+  do { \
+    struct curl_blob *st_blob_init = (struct curl_blob *)(structblob); \
```

Comment:
I used size_t for flag to have same type than len.

On my library, I choose type for two thing : alignement and less different type (this can be more easy for binding on another langage).
4 byte in more or less is blob structure will not waste computer memory.


But this is just my personnal experience, tell me the type you want for flag (if we use flag) :-)

Reply:
I switched to int flags (they are other int flags in curl), and removed macro.